app/actions/remote/entry/common.ts

```diff
@@ -125,21 +128,37 @@ export const entry = async (serverUrl: string, teamId?: string, channelId?: stri
 
     const models = await Promise.all(modelPromises);
 
-    return {models: models.flat(), initialChannelId, initialTeamId, prefData, teamData, chData, meData};
+    return {models: models.flat(), initialChannelId, initialTeamId, prefData, teamData, chData, meData, shouldPopToRoot};
 };
 
-export const fetchAppEntryData = async (serverUrl: string, since: number, initialTeamId = ''): Promise<AppEntryData | AppEntryError> => {
+export const fetchAppEntryData = async (serverUrl: string, sinceArg: number, initialTeamId = ''): Promise<AppEntryData | AppEntryError> => {
     const database = DatabaseManager.serverDatabases[serverUrl]?.database;
     if (!database) {
         return {error: `${serverUrl} database not found`};
     }
-
+    let since = sinceArg;
+    let shouldPopToRoot = false;
     const includeDeletedChannels = true;
     const fetchOnly = true;
 
     const confReq = await fetchConfigAndLicense(serverUrl);
     const prefData = await fetchMyPreferences(serverUrl, fetchOnly);
     const isCRTEnabled = Boolean(prefData.preferences && processIsCRTEnabled(prefData.preferences, confReq.config));
+    if (prefData.preferences) {
+        const crtToggled = await queryHasCRTChanged(database, prefData.preferences);
+        if (crtToggled) {
+            const currentServerUrl = await DatabaseManager.getActiveServerUrl();
+            const isSameServer = currentServerUrl === serverUrl;
+            if (isSameServer) {
+                shouldPopToRoot = true;
+                since = 0;
+            }
+            const {error} = await truncateCrtRelatedTables(serverUrl);
+            if (error) {
+                return {error: `Resetting CRT on ${serverUrl} failed`};
```

Comment:
What happens after this?

Reply:
I wanted to discuss about that... 
Should we add a retry with a circuit breaker?
Inform the user of what happened, since the experience would be broken then on?

Reply:
Yeah, I just don't know.

We should start a discussion in the channel, bring UX and PM on board to see what the experience should be.

Please add a comment about how in/probable is this to happen if you know. I'll check over the internet to see how common is for a query to truncate tables to fail

Reply:
The more I think about it, crashing the app should be the way to go.
I have no idea how common scenario that would be, but crashing would solve the problem

Reply:
Hahaha i like the way you think, but crashing should never be a solution I think

Reply:
I don't expect this to happen so we can keep it like this for now, in case we start seeing problems with this, we can figure out a solution for it.